lib/hash.rb

```diff
@@ -18,6 +18,12 @@ def squash
     end
   end
 
+  def remap_keys(remap)
+    change_keys do |key|
+      remap.key?(key) ? remap[key] : key
```

Comment:
the performance would not be better iterating over remap instead of change_keys?